### ISSUE TITLE
Remove aria-controls and aria-expanded from modal disclosures

### ISSUE THREAD

**Issue by dkempner** [ariakit/ariakit#925]
### Current behavior

Today, using `useDialogDisclosure` will return these attributes.

### Steps to reproduce the bug

Any `DialogDisclosure` adds these necessary attributes.

### Expected behavior

A button that opens a modal dialog is different because a user will only ever active the button to "expand" the modal dialog. The modal trigger button in WAI-ARIA's [Modal Dialog Example](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html) ("Add Delivery Address" button) also does not have the aria-expanded and aria-controls attributes. 

### Possible solutions

In my codebase I've started using a wrapper hook:

```ts
export const useModalDisclosure = (...props: Parameters<typeof useDialogDisclosure>) => {
  const final = useDialogDisclosure(...props)
  delete final['aria-controls']
  delete final['aria-expanded']
  return final
}
```

### Environment

N/A
